examples/addr2line.rs

```diff
@@ -0,0 +1,208 @@
+extern crate gimli;
+extern crate getopts;
+
+use std::env;
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let mut opts = getopts::Options::new();
+    opts.optopt("e", "exe", "Set the input file name (default is a.out)", "<executable>");
+
+    let matches = opts.parse(&args[1..]).unwrap();
+    let file_path = matches.opt_str("e").unwrap_or("a.out".to_string());
+    let file = obj::open(&file_path);
+    if obj::is_little_endian(&file) {
+        symbolicate::<gimli::LittleEndian>(&file, &matches);
+    } else {
+        symbolicate::<gimli::BigEndian>(&file, &matches);
+    }
+}
+
+fn parse_uint_from_hex_string(string: &str) -> u64
+{
+    if string.len() > 2 {
+        if string.starts_with("0x") {
+            return u64::from_str_radix(&string[2..], 16).expect("Failed to parse address");
+        }
+    }
+
+    return u64::from_str_radix(string, 16).expect("Failed to parse address");
+}
+
+fn entries_for_addresses<Endian>(file: &obj::File, addrs: &Vec<u64>) -> Vec<Option<gimli::DebugInfoOffset>>
```

Comment:
Nit: rename to `offsets_for_addresses` and use `offsets` instead of `dies` below. That would match the types and is less of a surprise to me, the reader.